students/Chris_Hallen/lesson03/mailroom.py

```diff
@@ -0,0 +1,72 @@
+        while donor_response == 'list':
+            print(donor_list[::3])
+            donor_response = input("Please enter a full name of a donor. Enter 'list' to show a full listing of all donors.\n> ")
+        member_index = donor_list.index(donor_response) if donor_response in donor_list else -1
+        if member_index >= 0 and member_index < len(donor_list):
+            donation_amount = 0
+            donation_amount = input("Please enter an amount that was donated: ")
+            donation_amount = float(donation_amount)
+            donor_list[member_index + 2] += 1
+            donor_list[member_index + 1] += donation_amount
+            print("\nThank you {} for donation number {} of ${:.2f}!".format(donor_list[member_index], donor_list[member_index + 1], donor_list[member_index + 2]))
+        else:
+            donation_amount = 0
+            donation_amount = input("Please enter an amount that was donated: ")
+            donation_amount = float(donation_amount)
+            donor_list.append(donor_response)
+            donor_list.append(donation_amount)
+            donor_list.append(1)
+            print("\nThank you {} for donation number {} of ${:.2f}!".format(donor_list[-3], donor_list[-1], donor_list[-2]))
+
+    def create_a_report(donor_list):
+        """This function prints a donor report with a formatted header and muliple lines thereafter for each donor"""
+        num_of_donors = len(donor_list) / 3
+        num_of_donors = int(num_of_donors)
+        print("Donor Name" + (' ' * 15) + " | " + "Total Given" " | " "Num Gifts" + " | " + "Average Gift\n" + ('- ' * 30))
+        donor_line = "{:25} ${:>11.2f} {:^13} ${:>11.2f}\n"
+        multiple_donor_lines = donor_line * num_of_donors
+
+        def add_average_donation(donor_list):
+            """This function within the create_a_report function returns a new list after taking the existing donor_list,
+               takes the first 3 entries of each donor - name, total donations, number of donations - and adds the
+               average amount each donor contributes to the end of each donor's profile.  This makes the donor_list 1/3 bigger
+               than the original list."""
+            new_length = (len(donor_list) / 3) * 4
+            longer_list = []
+            beg = 0
+            end = 3
+            while len(longer_list) < new_length:
+                longer_list.extend(donor_list[beg:end])
+                longer_list.append(longer_list[-2] / longer_list[-1])
+                beg += 3
+                end += 3
+            return longer_list
+        longer_list = add_average_donation(donor_list)
+        print(multiple_donor_lines.format(*longer_list))
+    response = input("Would you like to 'Send a Thank You', 'Create a Report' or 'quit'? \n > ")
+    response = response.lower()
+    while response != 'send a thank you' and response != 'create a report' and response != 'quit':
```

Comment:
it would be easier for a user to enter single letters or numbers as options:
1 - Send Thank You
or
"s" - Send Thank you